internal/pkg/store/ssm/ssm_test.go

```diff
@@ -77,11 +79,43 @@ func TestStore_ListProjects(t *testing.T) {
 			wantedProjectNames: nil,
 			wantedErr:          fmt.Errorf("broken"),
 		},
+		"with paginated response": {
+			mockGetParametersByPath: func(t *testing.T, param *ssm.GetParametersByPathInput) (output *ssm.GetParametersByPathOutput, e error) {
+				require.Equal(t, rootProjectPath, *param.Path)
+
+				// this closure references the `lastPageInPaginatedResp` variable
+				// in order to determine the content of the response.
+				if !lastPageInPaginatedResp {
+					lastPageInPaginatedResp = true
+					return &ssm.GetParametersByPathOutput{
+						Parameters: []*ssm.Parameter{
+							{
+								Name:  aws.String("/archer/chicken"),
+								Value: aws.String(testProjectString),
+							},
+						},
+						NextToken: aws.String("more"),
+					}, nil
+				}
+				return &ssm.GetParametersByPathOutput{
+					Parameters: []*ssm.Parameter{
+						{
+							Name:  aws.String("/archer/cow"),
+							Value: aws.String(cowProjectString),
+						},
+					},
+				}, nil
```

Comment:
I don't think it's worth changing as part of this PR, but could be worth considering moving to a similar strategy of generating `mockgen` mocks to clean this up a bit and be consistent with mocking elsewhere. It would help the mocks read a little easier and enable us to mock multiple calls, assert call orders, etc.

Reply:
It will definitely be awesome if we don't have to write the mocks!